File: Tree/BinaryBalancedTree/BalancedBinaryTree..py
"""
class TreeNode:
    def __init__(self, v):
        this.val = val
        this.left, this.right = None, None 
"""
class Solution:
    def balancedTree(self, root):
        if root is None:
            return false
            
        return self.maxDepth(root)
        
        
    def maxDepth(self, root):
        if root is None:
            return false;
            
        left = maxDepth(root.left)
        right = maxDepth(root.right)
        if left == -1 or right == -1 or abs(left - right) < 1:
            return false
        
        return true         

File: Tree/BinaryBalancedTree/BalancedBinaryTree..java
/*
class TreeNode {
	int val;
	TreeNode left, right;
	TreeNode(int v) {
		this.left = null;
		this.right = null;
		this.val = v
	}
}
*/
class Solution {
	public boolean balancedTree(TreeNode root) {
		if(root == null)
			return false;
		
		return (maxDepth(root) != -1);
	}
	private int maxDepth(TreeNode root) {
		if(root == null)
			return -1;
		
		int left = maxDepth(root.left);
		int right = maxDepth(root.right);
		if(left == -1 || right == -1 || abs{left - right} > 1) {
			return -1;
		}
		return (1 + Math.max(left, right));
	}
}